### PR TITLE
[flash_ctrl] Documentation updates

### PR DESCRIPTION
**Pull request by tjaychen**
- Addresses the questions from #10188 and #10189.

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

**Review by a-will — APPROVED** [review]

**Review by TimEwins — APPROVED** [review]

File: hw/top_earlgrey/ip/flash_ctrl/data/autogen/flash_ctrl.hjson [hunk 1]
```diff
@@ -423,8 +425,8 @@
             hwaccess: "hrw",
             name: "START",
             desc: '''
-              Start flash transaction.  This bit shall only be set after the other fields of the
-              CONTROL register and ADDR have been programmed
+              Start flash transaction.  This bit shall only be set at the same time or after the other
+              fields of the !!CONTROL register and !!ADDR have been programmed.
```
Comment (matutem) [inline hw/top_earlgrey/ip/flash_ctrl/data/autogen/flash_ctrl.hjson:429]:
```suggestion
              Start flash transaction.  This bit shall not be set before the other
              fields of the !!CONTROL register and !!ADDR have been programmed.
```

**Review by matutem — CHANGES_REQUESTED** [review]

**Review by alphan — APPROVED** [review]

File: hw/top_earlgrey/ip/flash_ctrl/data/autogen/flash_ctrl.hjson (hunk 1 above)
Comment (tjaychen) [reply hw/top_earlgrey/ip/flash_ctrl/data/autogen/flash_ctrl.hjson:429]: actually this one i probably won't change. The DV people looking at this specifically want the notion of "at the same time or after"

File: hw/ip/flash_ctrl/doc/_index.md [hunk 2]
```diff
@@ -219,6 +219,14 @@ When an RMA entry request is received from the life cycle manager, the flash con
 The flash controller then initiates RMA entry process and notifies the life cycle controller when it is complete.
 Unlike the seed phase, after the RMA phase, the flash controller does not grant control back to software as the system is expected to reboot after an RMA attempt.
 
+#### Initialization
+
+The flash protocol controller is initialized through {{< regref "INIT" >}}.
+When initialization is invoked, the flash controller requests the address and data scrambling keys from an external entity, [otp_ctrl]({{< relref "hw/ip/otp_ctrl/doc/_index.md#interface-to-flash-scrambler" >}}) in this case.
+
+After the scrambling keys are requested, the flash protocol controller then reads the root seeds out of the [secret partitions]({{< relref "#secret-information-partitions" >}}) and sends them to the key manager.
+Once the above steps are completed, the read buffers inside the flash physical controller are then enabled for operation.
```
[the remark this answers is not in the document]
Comment (tjaychen) [reply hw/ip/flash_ctrl/doc/_index.md]: updated

File: hw/ip/flash_ctrl/doc/_index.md [hunk 3]
```diff
@@ -219,6 +219,14 @@ When an RMA entry request is received from the life cycle manager, the flash con
 The flash controller then initiates RMA entry process and notifies the life cycle controller when it is complete.
 Unlike the seed phase, after the RMA phase, the flash controller does not grant control back to software as the system is expected to reboot after an RMA attempt.
 
+#### Initialization
+
+The flash protocol controller is initialized through {{< regref "INIT" >}}.
+When initialization is invoked, the flash controller requests the address and data scrambling keys from an external entity, [otp_ctrl]({{< relref "hw/ip/otp_ctrl/doc/_index.md#interface-to-flash-scrambler" >}}) in this case.
+
+After the scrambling keys are requested, the flash protocol controller then reads the root seeds out of the [secret partitions]({{< relref "#secret-information-partitions" >}}) and sends them to the key manager.
```
[the remark this answers is not in the document]
Comment (tjaychen) [reply hw/ip/flash_ctrl/doc/_index.md]: updated

**Comment by tjaychen** [comment]
@matutem please have a look and let me know.

File: hw/top_earlgrey/ip/flash_ctrl/data/autogen/flash_ctrl.hjson (hunk 1 above)
Comment (matutem) [reply hw/top_earlgrey/ip/flash_ctrl/data/autogen/flash_ctrl.hjson:429]: NP

**Review by matutem — APPROVED** [review]
LGTM

**Pull request merged**